src/riak_cs_config.erl

```diff
@@ -402,6 +403,15 @@ read_before_last_manifest_write() ->
 region() ->
     get_env(riak_cs, region, ?DEFAULT_REGION).
 
+%% @doc This switch changes mapreduce input from listkeys to
+%% fold_objects after riak_kv's mapreduce input optimization
+%% merged and enabled. With fold_objects it will have 1/2 IO
+%% when scanning the whole bucket.
+-spec use_2i_for_storage_calc() -> boolean().
+use_2i_for_storage_calc() ->
+    riak_cs_list_objects_utils:fold_objects_for_list_keys()
+        andalso get_env(riak_cs, use_2i_for_storage_calc, false).
```

Comment:
What happens if one set this switch `true` when underlying riak_kv does not support riak_kv#1072?

If it does no harm, then setting default value to `true` is better. If not, current default value `false` is mandatory (unless we have 2.0.6 before cs 2.0 :zap: ).